src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/KeyService.java

```diff
@@ -202,15 +206,27 @@ private static String signerFaultCode(String detail) {
 
     static final String KEY_NOT_FOUND_FAULT_CODE = signerFaultCode(X_KEY_NOT_FOUND);
 
+    public void deleteKey(String keyId) throws KeyNotFoundException, ActionNotPossibleException,
+            GlobalConfOutdatedException {
+        try {
+            deleteKey(keyId, true);
+        } catch (UnhandledWarningsException e) {
+            // Since "ignoreWarnings = true", the exception is never thrown
```

Comment:
Swallowing exceptions should _almost_ never be done. Even if UnhandledWarningsException should indeed be impossible, wrapping it as RuntimeException is preferrable, in case there's some weird bug that _does_ cause UnhandledWarningsException, or if someone later changes code so that our current assumptions are not valid anymore. Even if unlikely, I've seen those two things happen, and then it may mean a long bughunt since from outside everything looks like the key delete was successful.

Adding `throw new RuntimeException(e);` here also does not have any real downside, in case we are correct and this never happens. Comment is a good one to keep IMO, to explain this to updater.

Reply:
Good points, `RuntimeException` is now thrown in case `UnhandledWarningsException` is catched.